File: Selenium/Activity1.java
package activities;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import java.time.Duration;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class Activity1 {
    public static void main(String[] args) {
        WebDriverManager.chromedriver().setup();
        WebDriver driver = new ChromeDriver();

        driver.get("https://training-support.net");

        System.out.println("Title of the page is: " + driver.getTitle());

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
        WebElement aboutLink = wait.until(ExpectedConditions.elementToBeClickable(By.linkText("About")));
        aboutLink.click();

        System.out.println("Title after click is: " + driver.getTitle());

        driver.quit();
    }
}

File: Selenium/activity1.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from webdriver_manager.chrome import ChromeDriverManager
import time

# Setup ChromeDriver automatically
driver = webdriver.Chrome(service=Service(ChromeDriverManager().install()))

# Open the training site
driver.get("https://training-support.net")

# Print the page title
print("Title of the page is:", driver.title)

# Wait for About link to be clickable and click
wait = WebDriverWait(driver, 10)
about_link = wait.until(EC.element_to_be_clickable((By.LINK_TEXT, "About")))
about_link.click()

# Print the new page title
print("Title after click is:", driver.title)

# Wait a few seconds before closing
time.sleep(3)

driver.quit()
